crates/libtiny_tui/src/key_map.rs

```diff
@@ -0,0 +1,278 @@
+use std::collections::HashMap;
+use std::fmt::Display;
+
+use serde::de::{MapAccess, Unexpected, Visitor};
+use serde::{Deserialize, Deserializer};
+use term_input::{Arrow, Key};
+
+#[derive(Debug, PartialEq)]
+pub(crate) struct KeyMap(HashMap<Key, KeyAction>);
+
+#[derive(Debug, Copy, Clone, Deserialize, PartialEq)]
+#[serde(rename_all = "snake_case")]
+pub(crate) enum KeyAction {
```

Comment:
A `KeyAction` is dropped in any where a hardcoded `Key` combo was (except in a few areas where it makes no sense, like up arrow for scrolling back in history.